### DISCUSSION TITLE
[Feature] Mapview with a timeline view on the web (same as the mobile app)

### DISCUSSION THREAD

**Discussion by radh21301** [immich-app/immich#4519] [Feature Request]
### The feature

The current map view o the mobile is wonderful! I am refering to the way a timeline view pops up and updates as we move across the map. This is extremely helpful and really intuitive. It would be really nice if we can make the web version similar to that. Maybe something like this (this is a screenshot from a reddit post for photoprism). Though I think heatmaps (like on the immich mobile app) work better than those bubble markers.
![image](https://github.com/immich-app/immich/assets/45341381/2d385399-ff1f-4953-8711-97585fd2ea9e)

### Platform

- [ ] Server
- [X] Web
- [ ] Mobile

**Comment by hiveai** [comment]
I think there should be an easy way to click on a circle on the map and get to a grid view that fills the full screen (just as you get when you click on a place in the Explore view).

**Reply by brent20** [reply]
Came here to file a feature request for exactly this. Thanks!

**Comment by aisbergde** [comment]
Great feature request. Currently, I try to add assets, which I found on the map, to albums. This is not so easy, because each item needs to be assigned one by one. If we had this thumbnail view and if we could multiselect assets in this view and if we could do, what we can do in the timeline, this would be great.

In other words: **I would like to use a map to filter the timeline.**
And I prefer the count bubbles, as it is now. But this could be just a switch, if other users prefer the heatmap, why not.

**Reply by bjrne** [reply]
I agree, I also prefer the bubbles.

**Comment by chengcheng300** [comment]
Yes, idealy click on a map should be like open an Album, and my intention is to select them and create a Album as well. 

This is very useful for travel photos in remote areas.

**Reply by axgd-code** [reply]
I agree, this view could be the beginning of a new type of album with GPX, similar to Polarsteps. 

However, Immich is still missing a few features:
- A function to add notes to albums https://discord.com/channels/979116623879368755/1134549197379616859/1134551451478269972
- A display of tracks connecting the GPX points of the photos
- Tags on pictures


www.polarsteps.com/MarineGh/10380850-mon-1er-voyage-en-solo-en-colombie
<img width="1470" alt="image" src="https://github.com/user-attachments/assets/16c01fe9-0742-4cdd-b869-1587932c23f5">

https://github.com/immich-app/immich/discussions/6114

**Comment by jakub-sekula** [comment]
I agree this would be a great addition, something like the iCloud photos implementation where clicking a thumbnail on the map opens an "album" with all photos from that spot would make sorting through large libraries so much easier.

**Comment by dgtal1** [comment]
Yes! I started with immich today and have just posted the same feature request (oops!). I like how Photoprism does it. It's more convenient. Hope someone can pick tthis one and implement it :)

**Comment by tenephor** [comment]
This is the feature I'm waiting for in order to switch to immich.

**Comment by kaibagley** [comment]
I use this feature a lot on mobile and I'd absolutely love it if I could have a reactive panel of photos based on map view on web too!! I prefer the heatmap like on mobile

**Comment by jojo-gfx** [comment]
Ups, was also post the same request... 😅
Good to see, work is in progress... 👍

**Comment by ldivl** [comment]
I also like the heatmap concept more, show photos that are based on your zoom level in timeline.  However the this feature on the mobile app are basically non functional for me with 150k < photos.  The web version works fine for me, so performance would need to be top of mind for large libraries.

**Comment by dgtal1** [comment]
The heatmap looks fancy, but I have a feeling it has a negative impact on performance. And it doesn't really add much value IMO. In the map view you first of all want to see photos taken in a specific area and that's what this feature request is about. You know where you made them and if there was 1 photo or much more unless you're looking at a shared library. So I think if the heatmap is fancy but slows down the rendering and search performance - it should not be [art of the mobile app, where performance & battery life is first priority.

**Reply by aisbergde** [reply]
I agree, the heat map is a big performance killer and therefore not really usable, if you have a lot of assets. I would like to see the "number of assets" map view in the mobile app.

**Reply by TheNixie** [reply]
It's pretty - but the bubble lends itslef to the functionality of 'display all photos represented in the bubble' while the heat map is - pretty. :)

**Comment by jojo-gfx** [comment]
Would think the simple method of sized circles could be less stressful for the phone.

**Comment by grantfischer** [comment]
Any progress on this feature? Is this planned for 2025? Really seems needed for making the Map usable.

**Reply by YarosMallorca** [reply]
There is a pull request for it: https://github.com/immich-app/immich/pull/16494

**Reply by jbcodemonkey** [reply]
So glad to hear this. Right now, clicking a bundle of assets on the map returns them by most recent first. That makes sense in a lot of cases, but for viewing vacation photos, it's the wrong order. With the timeline, it would be easy to start at the beginning and work back to the present. Another interesting option would be a sort order button to reverse the set.

**Comment by andreymaznyak** [comment]
I want implement this features(support gpx timelines), in next month maybe.

I think basic implementation - load daily gpx track to db. And attach it to dates in timeline.

And the more complex implementation - sepatate component with custom date range.

**Reply by bo0tzz** [reply]
@andreymaznyak to clarify, this feature request doesn't involve GPX/tracks. It's just about having a grid view for the area that's visible on the map. I don't think we want gpx support in Immich directly; there's already good tools that do that, such as dawarich (which has an Immich integration).

**Reply by andreymaznyak** [reply]
ooo, thank you. I'm will try to up dawarich, with my immich instance.

**Reply by andreymaznyak** [reply]
maybe we should think about some ui integration of this tools

**Comment by Fifou2202** [comment]
+1 – I strongly support this feature.

I often use the map to find photos taken at very specific locations using GPS metadata. Right now, clicking on a location only shows one photo, and I have to swipe through manually.

It would be incredibly useful to view all photos in a gallery/grid mode, especially to bulk delete pictures from precise spots (e.g., test shots, duplicates, random locations). Much more accurate than filtering by city name. Thanks!

**Reply by bo0tzz** [reply]
@Fifou2202 please don't spam the same comment on all the dupes you can find...

**Comment by caseycarlson11** [comment]
Does anyone know if their is plans to implement this to web view? Seams strange that you would be able to see gallery on your device and not on the web. I would want to be able to view gallery of a location on the map, select them all and put them in an album.

**Comment by goldenman-kr** [comment]
I recommend to refer QuMagie web interface.
When I click number of photos on the map, it shows album list as with temporary tag instead of first photo.

![image](https://github.com/user-attachments/assets/6b71b8b6-fb37-4169-959f-14cc002990b2)
![image](https://github.com/user-attachments/assets/16358e12-1b0a-4d99-bc6f-6db80197a211)

I think this is good reference and simple to support this feature reusing existing modules.

**Comment by afeblot** [comment]
I also came here to suggest this, and happily found I'm not the only one seeing value in this feature :-)
(Thanks for Immich !)

**Comment by FunkyKwak** [comment]
I also support this, but please, do not erase the filter feature in the process, currently available on web only.
Being able to see the map with only favourite pictures is great !

**Comment by nikomiko** [comment]
Hope this will make its way to a future release as well.

**Comment by Idan37S** [comment]
That feature would be so cool,
I'm really using it a lot in the mobile app, and in the web in the pc it's really missing.

**Comment by liutang** [comment]
+1 for this feature. Clicking on a bubble in the map view brings up a random picture, which isn't useful. At a minimum, I was expecting something like Apple Photos where clicking on the numbered bubble brings up a listing of all the photos in that bubble.

**Comment by grantfischer** [comment]
+1 this imo is the only needed feature missing from web, and is the only thing that’s currently preventing me from switching over to use immich full time

**Comment by goldenman-kr** [comment]
thank you devs!

I found this is finally supportive since v2.6.0

Now immich is more perfect than ever.

Guys please update your server to the latest.

2026년 4월 9일 (목) 오전 3:05, grantfischer ***@***.***>님이 작성:

**Comment by pizzabackup** [comment]
Yes, great update, thank you!

One more upgrade would be that on the photo panel on the right, it automatically shows all photos that belong to the map area on the left.
If you zoom in or out, the photos on the right update to reflect the new map frame.

Wish this could be integrated!
